File: modules/playground/src/web_workers/router/background_index.dart
library playground.src.web_workers.router.background_index;

import "index_common.dart" show App;
import "dart:isolate";
import "package:angular2/platform/worker_app.dart";
import "package:angular2/core.dart";
import "package:angular2/platform/common.dart" show LocationStrategy, HashLocationStrategy;
import "package:angular2/src/web_workers/worker/router_providers.dart";

@AngularEntrypoint()
main(List<String> args, SendPort replyTo) {
  bootstrapApp(replyTo, App, [WORKER_APP_ROUTER, new Provider(LocationStrategy, useClass: HashLocationStrategy)]);
}

File: modules/playground/src/web_workers/router/background_index.ts
import {ROUTER_PROVIDERS} from '@angular/router-deprecated';
import {WORKER_APP_LOCATION_PROVIDERS} from '@angular/platform-browser';
import {bootstrapWorkerApp} from '@angular/platform-browser-dynamic';

import {HashLocationStrategy, LocationStrategy} from '@angular/common';
import {App} from './index_common';

export function main() {
  bootstrapWorkerApp(App, [
    ROUTER_PROVIDERS,
    WORKER_APP_LOCATION_PROVIDERS,
    {provide: LocationStrategy, useClass: HashLocationStrategy}
  ]);
}
